File: 02-alp-with-pos.py
def alphabet_position(text):
    pos = []
    for l in text:
      if ord(l) >= 65 and ord(l) <= 90:
        pos.append(str(ord(l) - 64))
      elif ord(l) >= 97 and ord(l) <= 122:
        pos.append(str(ord(l) - 96))
      else:
        continue
    return " ".join(pos)

print(alphabet_position("The sunset sets at twelve o' clock."))
print(alphabet_position("The narwhal bacons at midnight."))

File: 02-alp-with-pos.c
#include <stdio.h>
#include <string.h>

char *alphabet_position(char *text) {
  // Your code here
  char *res = NULL;
  res[0] = '\0';
  for(int i = 0; i < strlen(text); i++){
    if(text[i] >= 65 && text[i] <= 90) text[i] -= 64;
    else if(text[i] >= 97 && text[i] <= 122) text[i] -= 96;
    else continue;
    char str_num[20];
    sprintf(str_num, "%d", text[i]);
    int start_idx = strlen(res);
    int end_idx = start_idx + strlen(str_num);
    int j = 0;
    for(int k = start_idx; k < end_idx; k++){
      res[k] = str_num[j];
      j++;
    }
    res[end_idx] = ' ';
    res[end_idx + 1] = '\0';

    printf("%s\n", res);
  }
  printf("\n");
  return res;
}

int main(){
  char *ptr;
  ptr = alphabet_position("The sunset sets at twelve o' clock.");
  printf("%s\n", ptr);
}
